hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerConfiguration.java

```diff
@@ -486,4 +487,27 @@ private String getOMAddrKeyWithSuffix(String serviceId, String nodeId) {
     return ConfUtils.addKeySuffixes(OMConfigKeys.OZONE_OM_ADDRESS_KEY,
         serviceId, nodeId);
   }
+
+  /**
+   * Re-initializing OMStorage with new nodeId should fail.
+   */
+  @Test
+  public void testChangeOMNodeId() throws Exception {
```

Comment:
Can we also test a restart and reinit scenario where the value is changed in the config to be different from the value in the VERSION file? This would more closely resemble the mistake a user would make that we are trying to catch.